### PR TITLE
 Support USDC withdraw from savings through PSMActions

### PR DESCRIPTION
**Pull request by yivlad**

### CODE REVIEW

File: packages/app/src/domain/psm-actions/redeem-and-swap/useRedeemAndSwap.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export interface UseRedeemAndSwapArgs {
 // @note: Redeem a specified amount of `savingsToken` from the `savingsToken`
 // for `dai` and swap for `gem` in the PSM. Use this if you want to withdraw everything.
 // @note: Assumes PSM swap rate between `dai` and `gem` is 1:1.
-export function UseRedeemAndSwap({
+export function useRedeemAndSwap({
```
Comment (oskarvu) [inline packages/app/src/domain/psm-actions/redeem-and-swap/useRedeemAndSwap.ts:23]: 👌

**Review by oskarvu — APPROVED** [review]

File: packages/app/src/features/actions/flavours/native-sdai-withdraw/useCreateNativeSDaiWithdrawHandler.ts [hunk 2]
```diff
@@ -15,20 +18,39 @@ export function useCreateNativeSDaiWithdrawHandler(
...
 ): ActionHandler {
   const { enabled, onFinish } = options
+  const isUSDCWithdraw = action.token.symbol === 'USDC'
 
-  const withdraw = useVaultWithdraw({
+  const daiWithdraw = useVaultWithdraw({
     vault: action.sDai.address,
-    assetsAmount: action.token.toBaseUnit(action.value),
-    enabled: enabled && action.method === 'withdraw',
+    assetsAmount: action.method === 'withdraw' ? action.token.toBaseUnit(action.value) : BaseUnitNumber(0),
```
Comment (krzkaczor) [inline packages/app/src/features/actions/flavours/native-sdai-withdraw/useCreateNativeSDaiWithdrawHandler.ts:25]: simplify this by introducing isWithdraw and isReedem vars

File: packages/app/src/features/actions/flavours/native-sdai-withdraw/useCreateNativeSDaiWithdrawHandler.ts [hunk 3]
```diff
@@ -15,20 +18,39 @@ export function useCreateNativeSDaiWithdrawHandler(
...
+  const daiRedeem = useVaultRedeem({
     vault: action.sDai.address,
-    sharesAmount: action.sDai.toBaseUnit(action.value),
-    enabled: enabled && action.method === 'redeem',
+    sharesAmount: action.method === 'redeem' ? action.sDai.toBaseUnit(action.value) : BaseUnitNumber(0),
+    enabled: enabled && action.method === 'redeem' && !isUSDCWithdraw,
     onTransactionSettled: onFinish,
   })
 
+  const usdcWithdraw = useWithdrawAndSwap({
```
Comment (krzkaczor) [inline packages/app/src/features/actions/flavours/native-sdai-withdraw/useCreateNativeSDaiWithdrawHandler.ts:36]: don't you think that we could use another flavour to simplify all of that? split between DAI and USDC?

Comment (yivlad) [reply packages/app/src/features/actions/flavours/native-sdai-withdraw/useCreateNativeSDaiWithdrawHandler.ts:38]: I think the code is not that bad for now - but we should definitely consider this when adding more complexity to the logic.

**Review by krzkaczor — APPROVED** [review]

**Pull request merged**